### ISSUE TITLE
Error on building

### ISSUE THREAD

**Issue by lxgnzlz05** [ilyakurdyukov/rk3528-tvbox#8]
Hi, i'm just building the image for vontar dq08 as the guide posted in README, but i get a error like this:
![image](https://github.com/ilyakurdyukov/rk3528-tvbox/assets/64931332/0db33e10-5016-4ac7-8350-3f36d805abe4)

I'm running it on Ubuntu Desktop 24.04 LTS on Virtual Box.

If anyone can help, it will be very grateful
Thanks for all

**Comment by ilyakurdyukov** [comment]
Looks like some new warnings from the new GCC that have turned into errors. As you can see, the return type differs. You need to fix it so that the types match.

I abandoned this project months ago and have an older compiler, so you will have to fix this error yourself. This problem is in the kernel code. This is from Rockchip's U-Boot code [here](https://github.com/rockchip-linux/u-boot/blob/63c55618fbdc36333db4cf12f7d6a28f0a178017/common/command.c#L504).

You need to put a patch for this U-Boot into `patch/u-boot/legacy/board_rk3528-tvbox` directory, `armbian-build` should use this patch for the next build.

**Comment by lxgnzlz05** [comment]
Thanks for your response, i invested and i have 0 idea how this works.

I dont know how to do it 😥 i'm not acquainted with this kind of enviorements

**Comment by ilyakurdyukov** [comment]
I think you shouldn't install Linux on TV-boxes if you can't even do that. Because there will be many difficulties there.

```console
$ git clone --depth=1 https://github.com/rockchip-linux/u-boot.git
$ cd u-boot
$ sed -i 's/enum command_ret_t/int/' common/command.c
$ git patch > ../new_gcc_fix.patch
```

Then put `new_gcc_fix.patch` where I said earlier.

> You need to put a patch for this U-Boot into `patch/u-boot/legacy/board_rk3528-tvbox` directory, `armbian-build` should use this patch for the next build.

**Comment by lxgnzlz05** [comment]
Hi!! Thanks alot of your later comment, it get me very util.

I dont understant this last steps, (i have builded the image)


Then copy this `.dtb` to the first sector of the Armbian image in `<image-boot-part>/dtb/rockchip/`. And update the `.dtb` name in `armbianEnv.txt`.

### Booting from USB

Update U-Boot in EMMC to U-Boot from an Armbian image built using these patches. It was built with `CONFIG_ROCKCHIP_USB_BOOT=y` in U-Boot defconfig.

* USB 3.x devices do not work. Because for some reason USB 3.x flash drives are only compatible with USB 2.0, and for some reason U-Boot only detects devices that support USB 1.1. USB 2.0 flash drives support USB 1.1. USB 3.x devices will work with Linux after booting.

An example of how to extract `uboot.img` from an Armbian image:

```
$ dd if=Armbian.img bs=1M count=4 skip=8 > uboot.img
```

And how to write this in EMMC:

```
$ rkflashtool w 0x4000 0x2000 < uboot.img
```

Afther this, i got done :)

Thank you for your time and patience. I would like to know if you have any links for donations; you have been very helpful.

**Comment by ilyakurdyukov** [comment]
I don't think this will ever be done for you. Because there are many problems ahead with it. Next you won't be able to boot it. The HDMI driver is completely broken (for this TV-box) in the latest legacy Armbian kernel. HDMI was not working well at the end of 2023, but now it doesn't work at all.

You can build an image for a new kernel, but you need to update the `.dtb` sources for it, and HDMI works well there, but onboard Ethernet doesn't work.

You can copy the `.dtb` into the first partition like this (replace `/dev/sdx` with your SD card device):

```console
$ sudo mkdir /mnt/loopimage
$ sudo mount /dev/sdx1 /mnt/loopimage
$ sudo cp rk3528-vontar-dq08.dtb /mnt/loopimage/dtb/rockchip/
$ sudo sed -i "s/\/.*\.dtb/\/rk3528-vontar-dq08.dtb/" /mnt/loopimage/armbianEnv.txt
$ sudo umount /dev/sdx1
```

**Comment by lxgnzlz05** [comment]
i have the modifiedfile.img, y flashed to sdcard with balena etcher // rufus and when i insert the sd to my vontar dq08, it does not do nothing jeje

**Comment by ilyakurdyukov** [comment]
What does it mean it do nothing? In order to boot from the card, you need to turn off Android and turn off the power, then connect the power to boot from the card. Armbian on kernel 5.10 is waiting for it to be configured via ssh over Ethernet. As I wrote, it's difficult, and you don't know Linux and Armbian well enough to handle it on your own. And this is half-baked anyway, even I couldn't fix everything.

**Comment by bausensi** [comment]
> I don't think this will ever be done for you. Because there are many problems ahead with it. Next you won't be able to boot it. The HDMI driver is completely broken (for this TV-box) in the latest legacy Armbian kernel. HDMI was not working well at the end of 2023, but now it doesn't work at all.
> 
> You can build an image for a new kernel, but you need to update the `.dtb` sources for it, and HDMI works well there, but onboard Ethernet doesn't work.
> 
> You can copy the `.dtb` into the first partition like this (replace `/dev/sdx` with your SD card device):
> 
> $ sudo mkdir /mnt/loopimage
> $ sudo mount /dev/sdx1 /mnt/loopimage
> $ sudo cp rk3528-vontar-dq08.dtb /mnt/loopimage/dtb/rockchip/
> $ sudo sed -i "s/\/.*\.dtb/\/rk3528-vontar-dq08.dtb/" /mnt/loopimage/armbianEnv.txt
> $ sudo umount /dev/sdx1

i managed to enable ethernet on my X88 mini 13 by studying the DTB from the stock firmware. i'm not well versed on the build process but i've created this overlay that goes on top of the mangopi DTB, which has working HDMI. mangopi uses the second ethernet device at &gmac1, while TV boxes use gmac0 so this just switches which interface is queried for.

save this to something like `overlay_tvbox.dts` and use `armbian-add-overlay <file>`
```
/dts-v1/;
/plugin/;
/ {
        compatible = "rockchip,mangopi-m28k", "rockchip,rk3528";

        fragment@0 {
                target = <&gmac0>;
                __overlay__ {
                        status = "okay";
                };
        };

        fragment@1 {
                target = <&gmac1>;
                __overlay__ {
                        status = "disabled";
                };
        };
};
```

someone should eventually create a proper Armbian build file for TV boxes where everything works out of the box, in the meantime this stopgap should give you Ethernet.
